### ISSUE TITLE
[vim9class] overruled method doesn't require "signature exactly the same"

### ISSUE THREAD

**Issue by errael** [vim/vim#13255]
### Steps to reproduce

The following _doesn't_ get an error on the declaration of `Bar.Doit`
```
vim9script
    
class A
endclass

class B extends A
endclass

class C extends B
endclass
  
class Foo
  def Doit(p: B): B
    return B.new()
  enddef
endclass

class Bar extends Foo
  def Doit(p: B): C         # <<<<< spec violation, expect an error
    return B.new()          # <<<<< compilation error (not surprising)
  enddef
endclass

disa
```

### Expected behaviour

Compilation error for `Bar`'s `def Doit(p: B): C` since `C` is not "exactly the same" as `B`

A note
As expected, and desired, declaring Foo.Doit() to return a `B` does not prevent a `C` from being returned.
The following script outputs
```
object<Bar>
object<C>
```

```
class Bar extends Foo
  def Doit(p: B): B
    return C.new()
  enddef
endclass


var bar = Bar.new()
echo typename(bar)
var b = bar.Doit(B.new())
echo typename(b)
```

### Version of Vim

9.0.1976

### Environment

ubuntu

**Comment by errael** [comment]
> I thought we all agreed that the return value can be covariant type.

Not exactly my recollection, but I'm OK with that. I thought we'd noted that a covariant type could be returned, not that an overruled method could be declared to return a covariant type. It may not be quite as simple, but it's  commonly done as you suggest (isn't it?) and easy to explain. I will always wonder why the spec was written that way. Could it just be that he wanted 
```
return super.Doit(p)
```
to work in any subclass; as the final sentence of that paragraph suggests (it's not definitive)?

No idea if not  being able to return a `B` would matter anywhere in `vim9class`. Have no fear; if I notice something, I'll report it ;-)

Since the spec was returned to "exact same type", something needed fix'n. Note that `The signature ... exactly the same` is also at the end of the `Vim9-abstract-class` section.

**Comment by errael** [comment]
Closed by #13292
